app/assets/javascripts/delete_ideas.js

```diff
@@ -0,0 +1,11 @@
+var deleteIdeas = function(requestService) {
+  $('#ideas').on('click', '.delete', function(){
+    var idea = this.closest('.idea')
+    deleteIdeaInDatabase(idea, requestService).then($(idea).remove())
```

Comment:
Nailed it. I wasn't getting an error but it was executing immediately and not giving time for everything else.